OnLand/ControlLaptop/views.py

```diff
@@ -41,11 +42,22 @@ def inject_now():
 def submit_configuration():
     configuration_form = FlightConfigurationForm()
     if configuration_form.validate_on_submit():
-        configuration = validate_configuration_values(configuration_form)
-        if configuration['all_values_valid'] is True:
-            pod_communicator.send_configuration()
-            return jsonify({'status': 'ok'})
-        return jsonify({'error': configuration['error']})
+        FlightConfig.get_flight_config_instance().update_config(
+            {
+                'retrieval_timeout': int(configuration_form.retrieval_timeout.data),
+                'max_flight_time': int(configuration_form.max_flight_time.data),
+                'motor_speed': int(configuration_form.motor_speed.data),
+                'telemetry_port': int(configuration_form.telemetry_port.data),
+                'command_port': int(configuration_form.command_port.data),
+                'flight_length': int(configuration_form.flight_Length.data),
+                'heartbeat_timeout': int(configuration_form.heartbeat_timout.data),
+                'pod_address': configuration_form.pod_ip.data,
+                'pod_driver': 'Motor' if configuration_form.pod_driver.data is True else 'Simulation',
+            }
+        )
+        PodCommunicator.get_pod_communicator().send_configuration(
```

Comment:
Right now if we try and send a config and theres no connection it just throws an error on the flask side and nothing gets relayed to the front end. sendConfiguration should be able to tell us: success, failed to connect, BB failed to prase. So we can then have descriptive failures on the front end. 

Maybe we create the PodCommunicator object when the app starts but we take the _connect_to_pod() method out of the constructor and make it its own call. The Flask App will likely be on well before the pod is on. When we load the profile page we try and connect and relay that info with the page. Then we have a button on the profile page that we can click and it simply runs the podCommunicator.connect() function and returns if it successfully connected. Or if we submit a profile it obvs tries to connect before sending.

It'd be a v simple add on and we can get a better idea of if the submission is going to be successful before we do submit.